src/charts/line/line-chart.directive.js

```diff
@@ -0,0 +1,204 @@
+         $scope.data.yData0.push(Math.round(Math.random() * 100));
+         $scope.data.yData1.push(Math.round(Math.random() * 50));
+       };
+     });
+   </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfLineChart', function (pfUtils) {
+  'use strict';
+  return {
+    restrict: 'A',
+    scope: {
+      config: '=',
+      chartData: '=',
+      showXAxis: '=?',
+      showYAxis: '=?'
+    },
+    replace: true,
+    templateUrl: 'charts/line/line-chart.html',
+    controller: ['$scope',
+      function ($scope) {
+
+        // Create an ID for the chart based on the chartId in the config if given
+        $scope.lineChartId = 'lineChart';
+        if ($scope.config.chartId) {
+          $scope.lineChartId = $scope.config.chartId + $scope.lineChartId;
+        }
+
+        /*
+         * Convert the config data to C3 Data
+         */
+        $scope.getLineData = function (chartData) {
+          var lineData  = {
+            type: 'area'
+          };
+
+          if (chartData && chartData.dataAvailable !== false && chartData.xData) {
+            lineData.x = chartData.xData[0];
+            // Convert the chartData dictionary into a C3 columns data arrays
+            lineData.columns = Object.keys (chartData).map (function (key) {
+              return chartData[key];
+            });
+          }
+
+          return lineData;
+        };
+
+        /*
+         * Setup Axis options. Default is to not show either axis. This can be overridden in two ways:
+         *   1) in the config, setting showAxis to true will show both axes
+         *   2) in the attributes showXAxis and showYAxis will override the config if set
+         *
+         * By default only line and the tick marks are shown, no labels. This is a line and should be used
+         * only to show a brief idea of trending. This can be overridden by setting the config.axis options per C3
+         */
+
+        if ($scope.showXAxis === undefined) {
+          $scope.showXAxis = ($scope.config.showAxis !== undefined) && $scope.config.showAxis;
+        }
+
+        if ($scope.showYAxis === undefined) {
+          $scope.showYAxis = ($scope.config.showAxis !== undefined) && $scope.config.showAxis;
+        }
+
+        $scope.defaultConfig = $().c3ChartDefaults().getDefaultLineConfig();
+        $scope.defaultConfig.axis = {
+          x: {
+            show: $scope.showXAxis === true,
+            type: 'timeseries',
+            tick: {
+              format: function () {
+                return '';
+              }
+            }
+          },
+          y: {
+            show: $scope.showYAxis === true,
+            tick: {
+              format: function () {
+                return '';
+              }
+            }
+          }
+        };
+
+        // Setup the default configuration
+        $scope.defaultConfig.point.r = 1;
```

Comment:
This is set by the default patternfly config to be 3 and shouldn't be overridden here.